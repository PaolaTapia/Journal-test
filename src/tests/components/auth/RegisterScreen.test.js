import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RegisterScreen } from '../../../components/auth/RegisterScreen';
import configureStore from 'redux-mock-store'; 
import thunk  from "redux-thunk";
import { types } from '../../../types/types';
//import {startGoogleLogin} from '../../../actions/auth'; 

/*jest.mock('../../../actions/auth', ()=>({
     startGoogleLogin: jest.fn(), 
     startLoginEmailPassword: jest.fn()
}))*/

const middlewares = [thunk];
const mockStore = configureStore(middlewares); 

const initState={
     auth: {},
     ui: {
          loading: false,
          msgError: null
     }
}

let store= mockStore(initState); 
//store.dispatch=jest.fn();
//initState2= store.getState(); 
const wrapper= mount(
     <Provider store={store}>
          <MemoryRouter>
               <RegisterScreen />
          </MemoryRouter>
     </Provider>     
); 

describe('Pruebas en <RegisterScreen/>', () => {
     test('debe de mostrarse correctamente', () => {
          expect(wrapper).toMatchSnapshot(); 
     }); 

     test('debe de hacer el dispatch de la accion respectiva ', () => {
          const emailField=wrapper.find('input[name="email"]'); 

          emailField.simulate('change', {
               target:{
                    value:'',
                    name: 'email'
               }
          })
          wrapper.find('form').simulate('submit', {preventDefault(){}});
          const actions=store.getActions(); 
          //console.log(actions); 
          expect(actions[0]).toEqual({
               type: types.uiSetError, 
               payload: 'Email is not valid'
          });
     }); 
     test('debe de mostrar la caja de alerta con el error ', () => {
          const initState={
               auth:{},
               ui: {
                    loading: false, 
                    msgError: 'Email no es correcto'
               }
          }; 
          let store= mockStore(initState); 
          store.dispatch=jest.fn();

          const wrapper= mount(
          <Provider store={store}>
               <MemoryRouter>
                    <RegisterScreen />
               </MemoryRouter>
          </Provider>
          ); 

          expect(wrapper.find('.auth__alert-error').exists()).toBe(true); 
          expect(wrapper.find('.auth__alert-error').text().trim()).toBe(initState.ui.msgError); 


     })
     
     
     
})
