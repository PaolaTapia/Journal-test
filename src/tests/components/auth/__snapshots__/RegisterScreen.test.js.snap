// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RegisterScreen/> debe de mostrarse correctamente 1`] = `
Array [
  <h3>
     Register 
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate_faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Paola"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="mail@gmail.com"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="123456"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm"
      type="password"
      value="123456"
    />
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
       Register
    </button>
    <hr />
    <div>
      <a
        className="link mt-5"
        href="/auth/login"
        onClick={[Function]}
      >
         Already registered?
      </a>
    </div>
  </form>,
]
`;
